File: app/src/main/java/paul/barthuel/go4lunch/ui/list_view/RestaurantInfoDiffCallBack.java
package paul.barthuel.go4lunch.ui.list_view;

import androidx.annotation.NonNull;

class RestaurantInfoDiffCallBack extends androidx.recyclerview.widget.DiffUtil.ItemCallback<RestaurantInfo> {

    @Override
    public boolean areItemsTheSame(@NonNull RestaurantInfo oldItem, @NonNull RestaurantInfo newItem) {
        return oldItem.getId().equals(newItem.getId());
    }

    @Override
    public boolean areContentsTheSame(@NonNull RestaurantInfo oldItem, @NonNull RestaurantInfo newItem) {
        return oldItem.getName().equals(newItem.getName())
                && oldItem.getAddress().equals(newItem.getAddress())
                && oldItem.getDistance().equals(newItem.getDistance())
                && oldItem.getAttendies().equals(newItem.getAttendies())
                && oldItem.getOpeningHours() != null
                && oldItem.getOpeningHours().equals(newItem.getOpeningHours());
    }
}

File: app/src/main/java/paul/barthuel/go4lunch/ui/list_view/RestaurantInfoDiffCallBack.kt
package paul.barthuel.go4lunch.ui.list_view

import androidx.recyclerview.widget.DiffUtil

internal class RestaurantInfoDiffCallBack : DiffUtil.ItemCallback<RestaurantInfo>() {
    override fun areItemsTheSame(oldItem: RestaurantInfo, newItem: RestaurantInfo): Boolean {
        return oldItem.id == newItem.id
    }

    override fun areContentsTheSame(oldItem: RestaurantInfo, newItem: RestaurantInfo): Boolean {
        return oldItem.name == newItem.name
                && oldItem.address == newItem.address
                && oldItem.distance == newItem.distance
                && oldItem.attendies == newItem.attendies
                && oldItem.openingHours == newItem.openingHours
                && oldItem.backgroundColor == newItem.backgroundColor
    }
}